data-prepper-plugins/aws-lambda/src/main/java/org/opensearch/dataprepper/plugins/lambda/common/config/InvocationType.java

```diff
@@ -0,0 +1,52 @@
+package org.opensearch.dataprepper.plugins.lambda.common.config;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public enum InvocationType {
+    REQUEST_RESPONSE("request-response", "RequestResponse"),
+    EVENT("event", "Event");
+
+    private final String userInputValue;
+    private final String awsLambdaValue;
+
+    InvocationType(String userInputValue, String awsLambdaValue) {
+        this.userInputValue = userInputValue;
+        this.awsLambdaValue = awsLambdaValue;
+    }
+
+    public String getUserInputValue() {
```

Comment:
```suggestion
    @JsonValue
    public String getUserInputValue() {
```

This is needed to work with schema generation.